### DISCUSSION TITLE
anti-CSRF tokens for login forms

### DISCUSSION THREAD

**Discussion by jamesshuriff** [keycloak/keycloak#31315] [Ideas]
I'm maintaining a fork that adds an anti-CSRF <input> token to the login forms. Is this something anyone would be interested in having pulled into the repo?

**Comment by jonkoops** [comment]
We'll always welcome good security hardening improvements. If you would like to contribute one please log an issue for it and open a PR to resolve it. I would recommend reading our [contribution guidelines](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md).

**Comment by danielFesenmeyer** [comment]
@jamesshuriff:

Quote from the [keycloak docs](https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/threat/csrf.adoc):

> The OAuth 2.0 login specification requires that a state cookie matches against a transmitted state parameter. {project_name} fully implements this part of the specification, so all logins are protected.

I'm wondering what's still missing in Keycloak?

**Reply by danielFesenmeyer** [reply]
If you have found a suspected vulnerability, it should NOT be disclosed publicly. It should be reported to the Keycloak Security Team: https://www.keycloak.org/security
